manager/controlapi/network.go

```diff
@@ -174,6 +175,23 @@ func (s *Server) RemoveNetwork(ctx context.Context, request *api.RemoveNetworkRe
 		}
 	}
 
+	s.store.View(func(tx store.ReadTx) {
+		tasks, err = store.FindTasks(tx, store.All)
+	})
+	if err != nil {
+		return nil, grpc.Errorf(codes.Internal, "could not find tasks using network %s", request.NetworkID)
+	}
+
+	for _, t := range tasks {
```

Comment:
@dongluochen Yes it is very expensive but I am afraid there is no other simple way at this point since `go-memdb` cannot create an index which satisfies an `OR` of multiple fields in an object(or a field which is a slice and we need an index which satisfies each entry in the slice).